### PR TITLE
[buck-yeh-bux-mariadb-client] New port

### PR DESCRIPTION
**Pull request by buck-yeh**
Loose-coupled Throw-On-Error C++20 wrapper classes and utilities of mysql/mariadb [Connector/C API](https://mariadb.com/kb/en/mariadb-connectorc-api-functions/)

- #### What does your PR fix?  
  Nothing.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Supported
  - x64-windows
  - x64-windows-static
  - x64-windows-static-md
  - x86-windows
  - x86-windows-static
  - x86-windows-static-md
 
  Yes! x64-linux is marked _fail_ because in the real world any up-to-date linux C++ compiler will build but not in CI build here.
  ~~~
  buck-yeh-bux:x64-linux=fail
  buck-yeh-bux-mariadb-client:x64-linux=fail
  ~~~

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/buck-yeh-bux-mariadb-client/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "buck-yeh-bux-mariadb-client",
+  "version": "1.0.1",
+  "description": "Loose-coupled throw-on-error C++20 wrapper classes and utilities over mysql/mariadb Connector/C API",
+  "homepage": "https://github.com/buck-yeh/bux-mariadb-client",
+  "supports": "!(arm | uwp | osx | linux)",
```
Comment (JonLiu1993) [inline ports/buck-yeh-bux-mariadb-client/vcpkg.json:6]: @buck-yeh ,Thanks for your pr, this new port only supports "windows"?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/buck-yeh-bux-mariadb-client/vcpkg.json (hunk 1 above)
Comment (buck-yeh) [reply ports/buck-yeh-bux-mariadb-client/vcpkg.json:6]: > @buck-yeh ,Thanks for your pr, this new port only supports "windows"?

Affirmative

Comment (JonLiu1993) [reply ports/buck-yeh-bux-mariadb-client/vcpkg.json:6]: So I suggest you modify supports to "supports": "windows",

**Comment by dg0yt** [comment]
May I ask again it this project is already used by third-party projects? I.e. is there a demand for this port?

(Background: I'm experiencing slow reviews of updates of widely used packages. Adding new ports and dependencies doesn't improve this situation. Even less if we can't test dependencies on linux or osx.)

**Comment by buck-yeh** [comment]
> May I ask again it this project is already used by third-party projects? I.e. is there a demand for this port?
> 
> (Background: I'm experiencing slow reviews of updates of widely used packages. Adding new ports and dependencies doesn't improve this situation. Even less if we can't test dependencies on linux or osx.)

No other third party project is using this project. But I will dig into the causes why I previously built `buck-yeh-bux`, a dependee of this project, successfully using ArchLinux/vcpkg but failed to pass the checks in PR. That's why I rule linux out _for now_!

**Comment by buck-yeh** [comment]
Where can I get this log [`/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/install-x64-linux-dbg-out.log`](https://dev.azure.com/vcpkg/public/_build/results?buildId=63326&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172&l=1924) from details of x86-linux check?

**Comment by JonLiu1993** [comment]
> Where can I get this log [`/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/install-x64-linux-dbg-out.log`](https://dev.azure.com/vcpkg/public/_build/results?buildId=63326&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172&l=1924) from details of x86-linux check?

Here is the details log:
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/7567325/install-x64-linux-dbg-out.log)

**Comment by buck-yeh** [comment]
Need [/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/install-x64-linux-dbg-out.log](https://dev.azure.com/vcpkg/public/_build/results?buildId=63388&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172&l=1928) to see what is going on in x64-linux build again.

**Comment by dg0yt** [comment]
> Need [/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/install-x64-linux-dbg-out.log](https://dev.azure.com/vcpkg/public/_build/results?buildId=63388&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172&l=1928) to see what is going on in x64-linux build again.

Follow the details links from GH to Azure Pipelines, then follow the links to your PR's CI runs artifacts, then go to the line with the failure log for x64-linux, then click the three dots on the right side and choose download.

**Comment by buck-yeh** [comment]
>

**Comment by buck-yeh** [comment]
> 

 I got all log files following your directions. Thank you!

In `install-x64-linux-dbg-out.log` I read:
~~~c++
[1/20] /usr/bin/c++  -I/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/../include/bux -fPIC -Wall -Wextra -Wshadow -Wconversion -Wno-parentheses -Wno-unknown-pragmas -g3 -Og -std=c++2a -g -MD -MT src/CMakeFiles/bux.dir/FileLog.cpp.o -MF src/CMakeFiles/bux.dir/FileLog.cpp.o.d -o src/CMakeFiles/bux.dir/FileLog.cpp.o -c /mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/FileLog.cpp
FAILED: src/CMakeFiles/bux.dir/FileLog.cpp.o 
/usr/bin/c++  -I/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/../include/bux -fPIC -Wall -Wextra -Wshadow -Wconversion -Wno-parentheses -Wno-unknown-pragmas -g3 -Og -std=c++2a -g -MD -MT src/CMakeFiles/bux.dir/FileLog.cpp.o -MF src/CMakeFiles/bux.dir/FileLog.cpp.o.d -o src/CMakeFiles/bux.dir/FileLog.cpp.o -c /mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/FileLog.cpp
In file included from /mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/../include/bux/FileLog.h:3,
                 from /mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/FileLog.cpp:1:
/mnt/vcpkg-ci/buildtrees/buck-yeh-bux/src/c9e66a4664-c1c802f90b.clean/src/../include/bux/SyncLog.h:4:10: fatal error: concepts: No such file or directory
    4 | #include <concepts>     // std::convertible_to<>, std::derived_from<>
      |          ^~~~~~~~~~
compilation terminated.
~~~

The C++ compiler accepts `-std=c++2a` but not `-std=c++20`, and does not recognize header `<concepts>` yet. This leaves me no choice but change back `vcpkg.json`'s `"supports"` setting to:
~~~json
"supports": "!(arm | uwp | osx | linux)",
~~~
at least fort now.

**Comment by dg0yt** [comment]
> The C++ compiler accepts `-std=c++2a` but not `-std=c++20`, and does not recognize header `<concepts>` yet. This leaves me no choice but change back `vcpkg.json`'s `"supports"` setting to:
> 
> ```json
> "supports": "!(arm | uwp | osx | linux)",
> ```
> 
> at least fort now.

No, linux seems to be supported with a modern compiler. vcpkg linux CI isn't modern. So the solution is to add this to the ci baseline list of failures.

**Comment by buck-yeh** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> After committing all other changes, the version database must be updated
> ```shell
> git add -u && git commit
> git checkout cbc22a396c88d098247fe7f79f0913a952e1f77d -- versions
> ./vcpkg x-add-version --all
> ```
> 
> Diff
> ```diff
> diff --git a/versions/b-/buck-yeh-bux.json b/versions/b-/buck-yeh-bux.json
> index c5766c9e..cde398a0 100644
> --- a/versions/b-/buck-yeh-bux.json
> +++ b/versions/b-/buck-yeh-bux.json
> @@ -5,6 +5,11 @@
>        "version": "1.6.3",
>        "port-version": 0
>      },
> +    {
> +      "git-tree": "c952bbb93f152d6db8e4bc8d0fbed4fb6e7ead2a",
> +      "version": "1.6.2",
> +      "port-version": 0
> +    },
>      {
>        "git-tree": "fa8f4e2afd3a69f94ed947a1dfc0dcb4006ae9c9",
>        "version": "1.6.0",
> ```

File: ports/buck-yeh-bux-mariadb-client/vcpkg.json (hunk 1 above)
Comment (buck-yeh) [reply ports/buck-yeh-bux-mariadb-client/vcpkg.json]: Change `supports` line of both `ports/buck-yeh-bux/vcpkg.json` & `ports/buck-yeh-bux-mariadb-client/vcpkg.json` to:
~~~json
"supports": "!(arm | uwp | osx)",
~~~
And mark in `ci.baseline.txt` as x64-linux fail instead sue to CI's x64-linux build doesn't know header `<concept>`:
~~~
buck-yeh-bux:x64-linux=fail
buck-yeh-bux-mariadb-client:x64-linux=fail
~~~

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Thanks for your hard working

**Review by BillyONeal — APPROVED** [review]
LGTM after build completes.

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**